pygfx/renderers/wgpu/rendertexture.py

```diff
@@ -243,4 +250,11 @@ def fragment_shader(
             w = kernel[abs(x)] * kernel[abs(y)]
             val += t_tex.sample(s_sam, texcoord) * w
 
+    # Add noise - adds a bit of a scenematic effect
+    # todo: use spirv builtin random number gen
+    xy = v_texcoord.xy
+    noise = fract(sin(xy @ vec2(12.9898, 78.233) + u_render.time) * 43758.5453)
+    noise = u_render.noise * noise
+    val += vec4(noise, noise, noise, 1.0)
```

Comment:
shouldn't post processing be set up as separate shader passes?

Reply:
Yes. I now made the post processing steps configurable, by creating a list of PostProcessingStep objects. Added an example that adds noise to a scene.

Another example is fog. That should be relatively easy now!